feature_reflect_array.go

```diff
@@ -27,6 +27,10 @@ type arrayEncoder struct {
 }
 
 func (encoder *arrayEncoder) Encode(ptr unsafe.Pointer, stream *Stream) {
+	if encoder.arrayType.Len() == 0 {
```

Comment:
Looking at this again I realized that since an array's length is a static part of the type, I could also move this into `encoderOfArray` to return a special `emptyArrayEncoder` instead of checking the length on each value. That would be faster and also fix some assumptions in `EncodeInterface` and `IsEmpty` that the array has a non-zero length.

I don't quite understand the `special optimization for interface{}` in `EncodeInterface` and when that case would get called. The output in that case is hard-coded to produce `[null]`, but it seems like that also ignores the original length of the array, even if the contents should all be `null`.

Reply:
emptyArrayEncoder  sounds like a good idea, can you make it happen?